### PR TITLE
deprecate binary states code-wise

### PR DESCRIPTION
**Pull request by foxriver76**
see #2048

### CODE REVIEW

**Comment by foxriver76** [comment]
So, we also need depreacte in javascript adapter and the docs of javascript adapter + the developer docs. Are there other places?

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 1]
```diff
@@ -9816,8 +9833,13 @@ export class AdapterClass extends EventEmitter {
      * @param options optional
      * @param callback
      *
+     * @deprecated Please use `writeFile` instead of binary states
      */
     setForeignBinaryState(id: unknown, binary: unknown, options: unknown, callback?: unknown): any {
+        this._logger.warn(
```
Comment (Apollon77) [inline packages/adapter/src/lib/adapter/adapter.ts:9839]: really warn? user wise ... lets better do info

**Review by Apollon77 — CHANGES_REQUESTED** [review]

**Comment by Apollon77** [comment]
List of adapters with Binary State code:
* ioBroker.cameras
* ioBroker.octoprint
* ioBroker.phantomjs
* ioBroker.sonos
* ioBroker.restapi
* ioBroker.web
* ioBroker.admin
* ioBroker.chromecast
* ioBroker.cloud
* ioBroker.iot
* ioBroker.javascript
* ioBroker.lovelace
* ioBroker.mqtt
* ioBroker.ring
* ioBroker.sayit
* ioBroker.socket-classes (Lib)
* ioBroker.socketio
* ioBroker.ws

File: packages/adapter/src/lib/adapter/adapter.ts (hunk 1 above)
Comment (foxriver76) [reply packages/adapter/src/lib/adapter/adapter.ts]: good point, changed

**Pull request merged**